app/plugin_commands.go

```diff
@@ -98,7 +98,12 @@ func (a *App) tryExecutePluginCommand(args *model.CommandArgs) (*model.Command,
 	trigger = strings.ToLower(trigger)
 
 	a.Srv.pluginCommandsLock.RLock()
-	defer a.Srv.pluginCommandsLock.RUnlock()
+	locked := true
+	defer func() {
```

Comment:
My apologies, but I was ambiguous in my proposal of a goroutine -- that was the solution client-side, in the plugin, just as a workaround.

Looking at this now, I think a simpler solution would be to search for a matching command, release the lock, and then invoke. There shouldn't be a race on the entries in the slice under contention, just the slice itself, so that should be safe.

Reply:
I _did_ implement your goroutine suggestion in the plugin. I did not use sync.Map here (your other suggestion) since it was not really applicable. I like ^^, will do now.

Reply:
Done, thanks for the suggestion! much cleaner.